applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -533,10 +533,19 @@ static void raw_send(struct k_work *work)
 			}
 
 			if (datamode_off_pending) {
+#if defined(CONFIG_SLM_DATAMODE_URC)
+				sprintf(rsp_buf, "\r\n#XDATAMODE: %d\r\n",
+					size_finish - quit_str_len);
```

Comment:
Is it correct that this response is meant only for when `quit_str` is part of the data to send? This will also be sent if `datamode_handler` fails when `quit_str` is not part of the data. Then this will respond with `size_send `- `quit_str_len`. In this case, should it return with `size_send` or an error instead?

Reply:
I get your point, but the question is more about whether SLM should quit data mode when sending to LTE fails, and what to do if it fails

Reply:
Ok, thanks for clarifying :)

Reply:
I'll improve data mode in next PR focusing on that abnormal handling